api/v1alpha4/clusterclass_webhook.go

```diff
@@ -44,17 +46,19 @@ var _ webhook.Defaulter = &ClusterClass{}
 // Default satisfies the defaulting webhook interface.
 func (in *ClusterClass) Default() {
 	// Default all namespaces in the references to the object namespace.
-	if len(in.Spec.Infrastructure.Ref.Namespace) == 0 {
+	if in.Spec.Infrastructure.Ref != nil && len(in.Spec.Infrastructure.Ref.Namespace) == 0 {
```

Comment:
Defaulting is called before any validations.
This seems to be happening before open api schema validations too. 
Hence adding nil safe here.

Was getting nil pointer error without this check.